File: størmer_numbers.py
#!/usr/bin/env python3
# https://www.geeksforgeeks.org/stormer-numbers/

# Python program to print Stormer numbers 
  
from __future__ import print_function 
  
# Function to find largest prime factor 
  
def maxPrimeFactors(n): 
    # Initialize the maximum prime factor 
    # variable with the lowest one 
    maxPrime = -1
  
    # Print the number of 2's that divide n 
    while n % 2 == 0: 
        maxPrime = 2
        n /= 2
  
    # n must be odd at this point, thus skip 
    # the even numbers and iterate only for 
    # odd integers 
    for i in range(3, int(n**0.5)+1, 2): 
        while n % i == 0: 
            maxPrime = i 
            n /= i 
  
    # This condition is to handle the case when 
    # n is a prime number greater than 2 
    if n > 2: 
        maxPrime = n 
  
    return int(maxPrime) 
  
# Function to generate Stormer Numbers 
  
def stormer(n): 
    i = 1
    # Stores the number of Stormer numbers found 
    count = 0
    while(count < n): 
        t = i * i + 1
        if maxPrimeFactors(t) >= 2 * i: 
            print(i, end =' ') 
            count += 1
        i += 1
  
# Driver Method 
  
if __name__=='__main__': 
    n = 10
    stormer(n) 

File: rust/størmer-numbers.rs
// Størmer numbers

fn max_prime_factor(mut n: i64) -> i64 {
    // initialize maximum prime factor variable with lowest value
    let mut max_prime = -1;

    while (n % 2) == 0 {
        max_prime = 2;
        n /= 2;
    }

    // at this point, n must be odd
    for i in (3..n.isqrt()+1).step_by(2) {
        while (n % i) == 0 {
            max_prime = i;
            n /= i;
        }
    }

    // handle the case when n is prime and greater than 2
    if n > 2 {
        max_prime = n;
    }

    return max_prime;
}

fn størmer(n: i64) -> () {
    // find the n first Størmer numbers
    let mut i = 1;
    // let mut størmer_numbers = [i64; 10];
    let mut count = 0;
    while count < n {
        let t = i * i + 1;
        if max_prime_factor(t) >= 2 * i {
            println!("{} ", i);
            // (append newfound Størmer number to størmer_numbers array)
            count += 1;
        }
        i += 1;
    }
    // return størmer_numbers;
}

fn main() {
//   println!("{}", max_prime_factor(100));
//   println!("{}", max_prime_factor(198));
//   println!("{}", max_prime_factor(101));
//   println!("{}", max_prime_factor(191));
//   println!("{}", max_prime_factor(144));
//   println!("{}", størmer(4));
   størmer(4);
   størmer(9);
   størmer(25);
   størmer(10000);
}

